File: 4 - Replace Conditional Logic with Strategy/C#/MPG.ReplaceConditionalLogicWithStrategy/MPG.ReplaceConditionalLogicWithStrategy.Before/CapitalStrategyAdvisedLine.cs
namespace MPG.ReplaceConditionalLogicWithStrategy.Before
{
    public class CapitalStrategyAdvisedLine : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            return loan.GetCommitment() * loan.GetUnusedPercentage() * loan.Duration() * RiskFactor(loan);
        }
    }
}

File: 5 - Form Template Method/Python/before/capital_strategy_advised_line.py
from before.capital_strategy import CapitalStrategy


class CapitalStrategyAdvisedLine(CapitalStrategy):
    def __init__(self):
        CapitalStrategy.__init__(self)

    def capital(self, loan):
        return loan.get_commitment() * loan.get_unused_percentage() * self.duration(loan) * self.risk_factor_for(loan)
